app/util/analytics/analytics.py

```diff
@@ -84,16 +87,31 @@ def is_finished(self):
 
     def is_compliant(self):
         message = 'OK'
-        compliant = (self.actual_duration >= MIN_DEFAULTS[self.app_type]['test_duration'] and
-                     self.concurrency >= MIN_DEFAULTS[self.app_type]['concurrency'])
+
+        if self.app_type == JSM:
+            compliant = (self.actual_duration >= MIN_DEFAULTS[self.app_type]['test_duration'] and
+                         self.concurrency >= MIN_DEFAULTS[self.app_type]['customer_concurrency']
+                         + MIN_DEFAULTS[self.app_type]['agent_concurrency'])
+        else:
+            compliant = (self.actual_duration >= MIN_DEFAULTS[self.app_type]['test_duration'] and
+                         self.concurrency >= MIN_DEFAULTS[self.app_type]['concurrency'])
+
         if not compliant:
             err_msg = []
             if self.actual_duration < MIN_DEFAULTS[self.app_type]['test_duration']:
                 err_msg.append(f"Test run duration {self.actual_duration} sec < than minimum test "
                                f"duration {MIN_DEFAULTS[self.app_type]['test_duration']} sec.")
-            if self.concurrency < MIN_DEFAULTS[self.app_type]['concurrency']:
-                err_msg.append(f"Test run concurrency {self.concurrency} < than minimum test "
-                               f"concurrency {MIN_DEFAULTS[self.app_type]['concurrency']}.")
+                if self.app_type == JSM:
+                    min_default_concurrency = MIN_DEFAULTS[JSM]['customer_concurrency'] + \
+                                              MIN_DEFAULTS[JSM]['agent_concurrency']
+                    if self.concurrency < min_default_concurrency:
+                        err_msg.append(f"Total test run concurrency {self.concurrency} < than minimum required test "
```

Comment:
This message and check logic can be improved to be checked separately not only as a sum of both. I suggest creating tech debt to improve/refactor concurrency validation logic in the future.